packages/web-pkg/src/composables/keyboardActions/useKeyboardActions.ts

```diff
@@ -15,8 +16,25 @@ export enum ModifierKey {
   Shift = 'Shift'
 }
 
-export const useKeyboardActions = (keyBindOnElementId: string | null = null) => {
-  const actions = ref([])
+export interface KeyboardActions {
+  actions: Ref<Array<KeyboardAction | null>>
+  actionsFiltered: Ref<Array<KeyboardAction>>
+  selectionCursor: Ref<number>
+  removeKeyAction: (index: number) => void
+  resetSelectionCursor: () => void
+  bindKeyAction: (keys: { primary: Key; modifier?: ModifierKey }, callback: () => void) => number
+}
+
+export interface KeyboardAction {
+  primary: Key
+  modifier: ModifierKey | null
+  callback: (event: KeyboardEvent) => void
+}
+
+export const useKeyboardActions = (keyBindOnElementId: string | null = null): KeyboardActions => {
+  const actions = ref<Array<KeyboardAction | null>>([])
+  const actionsFiltered = computed(() => actions.value.filter((action) => action !== null))
```

Comment:
I'm a bit confused by `actions` and `actionsFiltered`. So `actions` are all actions that have been registered, including those that might have been removed again? And `actionsFiltered` are only the actions that are currently registered?

Reply:
And you need the index to successfully remove actions again, right? What do you think of generating an id when registering a new action und use that instead of the index? We have a `uuid` lib that could do that.

I would feel a bit more robust IMO. Also you could use slice and we would get rid of `actionsFiltered`.

Reply:
It just feels more robust and also we would only have one action property that holds all the currently registered actions. So I prefer working with ids here, yes 👍

Reply:
Pushed should be good now.